src/OrchardCore/Orchard.Mvc.Core/ModularRazorPageFilter.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+using Orchard.Environment.Shell.Builders.Models;
+
+namespace Orchard.Mvc
+{
+    public class ModularRazorPageFilter : IPageFilter
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public ModularRazorPageFilter(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public void OnPageHandlerExecuted(PageHandlerExecutedContext context)
+        {
+            var shellBluePrint = _httpContextAccessor.HttpContext.RequestServices.GetRequiredService<ShellBlueprint>();
```

Comment:
Looks hacky.

Reply:
Yes, quickly done for testing, but look at `ShellFeatureApplicationPart` it uses exactly the same line of code.
I will commit a version which inject directly `IShellFeaturesManager` to get enabled features.